core/src/main/scala/kafka/server/FetchSession.scala

```diff
@@ -411,54 +436,40 @@ class IncrementalFetchContext(private val time: Time,
     }
   }
 
-  // Iterator that goes over the given partition map and selects partitions that need to be included in the response.
-  // If updateFetchContextAndRemoveUnselected is set to true, the fetch context will be updated for the selected
-  // partitions and also remove unselected ones as they are encountered.
-  private class PartitionIterator(val iter: FetchSession.RESP_MAP_ITER,
```

Comment:
The `iterator` is unnecessary since we have to generate `list` collection in order to calculate message size.